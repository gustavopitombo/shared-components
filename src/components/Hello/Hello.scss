h1.hello-world {
    color: green
}