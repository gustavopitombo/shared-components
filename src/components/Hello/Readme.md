Basic title:

```tsx
<Hello>Hello User</Hello>
```