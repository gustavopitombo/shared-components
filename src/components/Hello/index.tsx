import React from 'react';

export const Hello = (props : any) => (
    <h1 data-testid="shd-hello" className="hello-world">
        {props.children}
    </h1>
)

export default Hello;
