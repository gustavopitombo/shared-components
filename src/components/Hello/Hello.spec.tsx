import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Hello from './index';

describe("Hello component", () => {
    it("render button", () => {
        const { getByTestId } = render(<Hello>Hello User</Hello>);   
        expect(getByTestId("shd-hello")).toHaveTextContent("Hello User");
    });

    it("matches snapshot", () => {
        const tree = renderer.create(<Hello>Clique aqui</Hello>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})