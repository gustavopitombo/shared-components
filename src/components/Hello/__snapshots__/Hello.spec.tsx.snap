// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component matches snapshot 1`] = `
<h1
  className="hello-world"
  data-testid="shd-hello"
>
  Clique aqui
</h1>
`;
